src/app/utils/field-validation/index.ts

```diff
@@ -41,6 +52,41 @@ const doFieldTypesMatch = (
     : right(undefined)
 }
 
+/**
+ * Returns true if response appears on a hidden field.
+ * This may happen if a submission is made programatically to try and bypass form logic.
+ * @param response The submitted response
+ */
+const isResponsePresentOnHiddenField = (
+  response:
+    | ISingleAnswerResponse
+    | ICheckboxResponse
+    | ITableResponse
+    | IAttachmentResponse,
+): boolean => {
+  if (isProcessedSingleAnswerResponse(response)) {
+    if (!response.isVisible && response.answer.trim() !== '') {
+      return true
+    }
+  }
+  if (isProcessedCheckboxResponse(response)) {
+    if (!response.isVisible && response.answerArray.length > 0) {
+      return true
+    }
+  }
+  if (isProcessedTableResponse(response)) {
+    if (!response.isVisible && response.answerArray[0].length > 0) {
+      return true
+    }
+  }
+  if (isProcessedAttachmentResponse(response)) {
+    if (!response.isVisible && response.filename.trim() !== '') {
+      return true
+    }
+  }
+  return false
```

Comment:
i think we should be using an `if-else if-else` structure here for the avoidance of doubt

Reply:
edited